Sources/Networking/Operations/Handling/CustomerInfoResponseHandler.swift

```diff
@@ -32,7 +68,38 @@ class CustomerInfoResponseHandler {
             }
             .mapError(BackendError.networkError)
 
-        completion(result)
+        self.handle(result: result, completion: completion)
+    }
+
+    private func handle(
+        result: Result<CustomerInfo, BackendError>,
+        completion: @escaping CustomerAPI.CustomerInfoResponseHandler
+    ) {
+        guard self.offlineEntitlementsEnabled,
+              result.error?.isServerDown == true,
+              #available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *) else {
+            completion(result)
+            return
+        }
+
+        _ = Task<Void, Never> {
+            do {
+                completion(.success(try await self.computeOfflineCustomerInfo()))
```

Comment:
A separate thing but somewhat related, I think this will call the customer info listener with the offline customer info correct? But since we are not caching that value, how are you handling users calling `getCustomerInfo` after that? 

In android, we currently would just return the offline customer info in cache (in memory) first, then initiate another fetch from the backend. 

I could see an argument for being stateless and just make it so we don't have cached customer info after we "enter" offline entitlements mode, so if trying to get the value from cache, it would fail and we should just initiate a backend request always... This might be slower in some conditions though so not sure if it's a good idea... Personally, I think caching the offline customer info would mimick real behavior more closely, but I can understand that stateless could also help avoiding some issues...